eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/FieldEmpty.php

```diff
@@ -0,0 +1,201 @@
+     */
+    protected function getFieldsInformation($fieldIdentifier)
+    {
+        $fieldMapArray = [];
+        $fieldMap = $this->contentTypeHandler->getSearchableFieldMap();
+
+        foreach ($fieldMap as $contentTypeIdentifier => $fieldIdentifierMap) {
+            if (!isset($fieldIdentifierMap[$fieldIdentifier])) {
+                continue;
+            }
+
+            $fieldTypeIdentifier = $fieldIdentifierMap[$fieldIdentifier]['field_type_identifier'];
+            $fieldMapArray[$fieldTypeIdentifier]['ids'][] = $fieldIdentifierMap[$fieldIdentifier]['field_definition_id'];
+            if (!isset($fieldMapArray[$fieldTypeIdentifier]['column'])) {
+                $fieldMapArray[$fieldTypeIdentifier]['column'] = $this->fieldConverterRegistry->getConverter($fieldTypeIdentifier)->getIndexColumn();
+            }
+
+            $fieldType = $this->fieldTypeService->getFieldType($fieldTypeIdentifier);
+            $fieldMapArray[$fieldTypeIdentifier]['empty_value'] = $fieldType->getEmptyValue();
+        }
+
+        if (empty($fieldMapArray)) {
+            throw new InvalidArgumentException(
+                '$criterion->target',
+                "No searchable fields found for the given criterion target '{$fieldIdentifier}'."
+            );
+        }
+
+        return $fieldMapArray;
+    }
+
+    /**
+     * Generate query expression for a Criterion this handler accepts.
+     *
+     * accept() must be called before calling this method.
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotImplementedException If no searchable fields are found for the given criterion target.
+     *
+     * @param \eZ\Publish\Core\Search\Legacy\Content\Common\Gateway\CriteriaConverter $converter
+     * @param \eZ\Publish\Core\Persistence\Database\SelectQuery $query
+     * @param \eZ\Publish\API\Repository\Values\Content\Query\Criterion $criterion
+     * @param array $languageSettings
+     *
+     * @return \eZ\Publish\Core\Persistence\Database\Expression
+     *
+     * @throws \eZ\Publish\Core\Base\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\Core\Persistence\Legacy\Content\FieldValue\Converter\Exception\NotFound
+     */
+    public function handle(
+        CriteriaConverter $converter,
+        SelectQuery $query,
+        Criterion $criterion,
+        array $languageSettings
+    ) {
+        $fieldsInformation = $this->getFieldsInformation($criterion->target);
+
+        $subSelect = $query->subSelect();
+        $subSelect->select(
+            $this->dbHandler->quoteColumn('contentobject_id')
+        )->from(
+            $this->dbHandler->quoteTable('ezcontentobject_attribute')
+        );
+
+        $whereExpressions = [];
+
+        foreach ($fieldsInformation as $fieldTypeIdentifier => $fieldsInfo) {
+            if ($fieldsInfo['column'] === false) {
+                continue;
+            }
+
+            $filter = '';
+
+            if ($criterion->value[0] === Criterion\IsFieldEmpty::EMPTY) {
```

Comment:
How did we get an array here as `value`? TL;DR;

Reply:
Not sure - some internal magic I guess